### ISSUE TITLE
Unused variables do not always show a warning

### ISSUE THREAD

**Issue by gfour** [souffle-lang/souffle#855]
In the following example, Souffle only shows a warning for the second case where 'y' occurs once:

```
.decl A(a1:symbol, a2:symbol)
.output A
.decl B(b1:symbol, b2:symbol)
.output B
.decl C(c:symbol)
.output C

B("hello", "42").

C(x),
A(x, x) :- B(x, y).  // unused 'y' (case 1)                                                                                                                                                                                                                   

A(x, x) :- B(x, y).  // unused 'y' (case 2) 
```

```
$ souffle -c -o exe warnings.dl
Warning: Variable y only occurs once in file warnings.dl at line 13                                                            
A(x, x) :- B(x, y).  // unused 'y' (case 2)                    
----------------^---------------------------
$ souffle --version
Souffle: 1.5.1
```

This behavior seems to hide unused variables when they appear in rules with multiple-clause heads.

**Comment by azreika** [comment]
Good catch. This is because the parser expands the multiple heads into separate rules, which are marked as internally generated and so skipped by the semantic checker when looking for unused variables. The same happens with disjunctions for the same reason:

```
A(x,x) :- B(x,y) ; B(x,z). // unused 'y' and 'z'.
```

Related to issue #829.

**Comment by azreika** [comment]
Fixed by PR #1288.
